### PR TITLE
Added alert component

### LINKED ISSUE #343: Adding alerts as new UI component
I want to create an alert UI component like this screenshot.

### PR DESCRIPTION
**Pull request by ettiennelouw**
I added simple alert components with default values for info, success, warning & danger, and tried to cover all the examples on the tailwindUI page for alerts. I use the card components as more or less a template, I don't see any tests for cards so I did not do tests for the alert, should I?

Options:
- padding = string
- rounded = string
- border = bool
- shadow = bool
- set icon = string
- set heading = string
- set text = string
- actions = to insert buttons or something
- set alertClasses="" function to add more custom classes
- set dismiss action with event wire:click="dismiss"

### CODE REVIEW

**Comment by ettiennelouw** [comment]
```
<x-alert border alertClasses="my-2">
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid pariatur, ipsum similique veniam quo totam eius aperiam dolorum.
</x-alert>

<x-alert icon="user" alertClasses="my-2" padding="p-10"/>

<x-alert info shadow icon="information-circle" heading="Attention information" text="Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid pariatur, ipsum similique veniam quo totam eius aperiam dolorum." alertClasses="my-2"/>

<x-alert warning dismiss icon="exclamation" text="Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid pariatur, ipsum similique veniam quo totam eius aperiam dolorum." alertClasses="my-2" />

<x-alert success icon="information-circle" text="Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid pariatur, ipsum similique veniam quo totam eius aperiam dolorum." alertClasses="my-2"/>

<x-alert danger shadow border dismiss icon="x-circle" heading="Attention danger" text="Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid pariatur, ipsum similique veniam quo totam eius aperiam dolorum." alertClasses="my-2">
    <x-slot name="actions">
        <button type="button" class="bg-red-50 px-2 py-1.5 rounded-md text-sm font-medium text-red-700 hover:bg-red-100 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-green-50 focus:ring-green-600">View status</button>
        <button type="button" class="ml-3 bg-red-50 px-2 py-1.5 rounded-md text-sm font-medium text-red-700 hover:bg-red-100 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-green-50 focus:ring-green-600">Dismiss</button>
    </x-slot>
</x-alert>
```
Screenshot 2022-07-13 at 20 53 59
Screenshot 2022-07-13 at 20 53 53

**Comment by ph7jack** [comment]
@ettiennelouw Thanks for contributing. I'm glad about your interest. I'm reviewing this PR and creating the documentation.
Could you add some tests for this component?

**Comment by ettiennelouw** [comment]
Tests added if you can just verify and see if they are sufficient. Also made the icon be added by default if you choose info, warning, success and danger but you can override them by specifying a icon=""

**Review by ph7jack — CHANGES_REQUESTED** [review]
Could you refactor the Alert class as I commented in the avatar PR?

Could you create different components for each alert type?
Create a base alert and extends the base class to create other classes.
Small components are simple to maintain

File: src/View/Components/Alert.php [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        ?string $icon = null,
+        ?string $heading = null,
+        ?string $text = null,
+        ?string $actions = null,
+        bool $border = false,
+        bool $shadow = false,
+        bool $info = false,
+        bool $warning = false,
+        bool $success = false,
+        bool $danger = false,
```
Comment (ph7jack) [inline src/View/Components/Alert.php]:
```suggestion
        bool $error = false,
```

File: src/View/Components/Alert.php [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        bool $dismiss = false,
```
Comment (ph7jack) [inline src/View/Components/Alert.php]:
```suggestion
        bool $dismissable = false,
```

File: src/View/Components/Alert.php [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    public string $backgroundColor = 'bg-gray-50 dark:bg-secondary-700';
+
+    public string $subjectColor = 'text-gray-700 dark:text-gray-400';
+
+    public function __construct(
+        ?string $padding = 'p-4',
+        ?string $rounded = 'rounded-md',
+        ?string $icon = null,
+        ?string $heading = null,
+        ?string $text = null,
```
Comment (ph7jack) [inline src/View/Components/Alert.php]:
```suggestion
        ?string $title = null,
        ?string $message = null,
```

File: resources/views/components/alert.blade.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        </div>
+        @if($dismiss)
+            <div class="ml-auto pl-3">
+                <div class="-mx-1.5 -my-1.5">
+                    <button type="button" wire:click="dismiss" class="inline-flex {{ $backgroundColor }} rounded-md p-1.5 {{ $subjectColor }} focus:outline-none focus:ring-2 focus:ring-offset-2">
+                        <span class="sr-only">Dismiss</span>
+                        <x-icon name="x" class="h-5 w-5" />
+                    </button>
+                </div>
+            </div>
```
Comment (ph7jack) [inline resources/views/components/alert.blade.php]: Add a **wire:click** with a default action is bad for the reusability, many components can have different methods to handle the dismissal of an alert. Just a slot will be good, if the alert is dismissable a custom logic will be required to the Livewire Component to show/hide the alert.

File: src/View/Components/Alert.php (hunk 1 above)
Comment (ettiennelouw) [reply src/View/Components/Alert.php]: Updated it to be more in line with the buttons
public bool $info = false,
public bool $warning = false,
public bool $positive = false,
public bool $negative = false,

**Comment by ettiennelouw** [comment]
I updated the alerts and tests, split out the class to alert and basealert classes just not sure if it is worth it to split each into its own wont that create a lot of duplicates? Or I'm not 100% sure i understand :)

_Could you create different components for each alert type?
Create a base alert and extends the base class to create other classes.
Small components are simple to maintain_

**Comment by ph7jack** [comment]
> I updated the alerts and tests, split out the class to alert and basealert classes just not sure if it is worth it to split each into its own wont that create a lot of duplicates? Or I'm not 100% sure i understand :)
> 
> _Could you create different components for each alert type? Create a base alert and extends the base class to create other classes. Small components are simple to maintain_

You can continue using the same blade file for all alerts, but multiple classes with custom CSS implementation. This way will reduce the maintenance process

```
Components/Alerts/Info.php
Components/Alerts/Success.php
...

```


But it was my old idea, we can now still use one blade and one PHP class, ignore my the past message.
In one class will be possible to have all tailwind colors in the future.

File: src/config/wireui.php [hunk 5]
```diff
@@ -59,9 +59,9 @@
         |
      */
     'components' => [
-        'avatar' => [
```
Comment (ph7jack) [inline src/config/wireui.php:62]: rollback the avatar component

File: resources/views/components/alert.blade.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<div class="{{ $alertClasses }}">
```
Comment (ph7jack) [inline resources/views/components/alert.blade.php]:
```suggestion
<div {{ $attributes->class($alertClasses) }}>
```

File: resources/views/components/alert.blade.php [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    <div class="flex">
+
+        @if($icon)
```
Comment (ph7jack) [inline resources/views/components/alert.blade.php]:
```suggestion
        @if ($icon)
```

File: resources/views/components/alert.blade.php [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+                <p>{{ $message ?? $slot }}</p>
+            </div>
+            @if($actions)
+                <div class="mt-4">
+                    <div class="-mx-2 -my-1.5 flex">
+                        {{ $actions }}
+                    </div>
+                </div>
+            @endif
+        </div>
```
Comment (ph7jack) [inline resources/views/components/alert.blade.php]:
```suggestion
        <div class="ml-3">
            @if ($title)
                <h3 class="mb-2 text-sm font-semibold {{ $subjectColor }}">
                    {{ $title }}
                </h3>
            @endif

            <div class="text-sm {{ $subjectColor }}">
                <p>{{ $message ?? $slot }}</p>
            </div>
            
            @if ($actions)
                <div class="mt-4">
                    <div class="-mx-2 -my-1.5 flex">
                        {{ $actions }}
                    </div>
                </div>
            @endif
        </div>
```

File: resources/views/components/alert.blade.php [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+        @if($dismissible)
```
Comment (ph7jack) [inline resources/views/components/alert.blade.php]:
```suggestion
        @if ($dismissible)
```

**Review by ph7jack — CHANGES_REQUESTED** [review]

**Comment by ettiennelouw** [comment]
I should not have pulled main into my branch, Im closing this one and will make a new pull request.

**Pull request closed without merging**